modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -168,9 +182,17 @@ const DomParser = (settings?: DomParserSettings, schema = Schema()): DomParser =
 
         if (!isEmpty(schema, nonEmptyElements, whitespaceElements, newParent)) {
           parent.insert(newParent, parents[0], true);
-          parent.insert(node, newParent);
+          if (schema.isValidChild(parent.name, node.name)) {
+            parent.insert(node, newParent);
+          } else {
+            unwrapInvalidNode(node);
+          }
         } else {
-          parent.insert(node, parents[0], true);
+          if (schema.isValidChild(parent.name, node.name)) {
```

Comment:
This doesn't look correct? I believe it should be `parents[0]` instead as that's where it's attempting to insert it into.

Reply:
We are calling `.insert` on `parent` not on `parents[0]`, `parents[0]` is the `refNode` used by `insert` (i.e. the sibling we will be placing `node` next to). 

```
<parent>
  <------------ // where we want to put `<node></node>` (before `refNode` because we pass `true` as the 3rd argument)
  <parents[0]> // refNode
  </parents[0]>
</parent>
```

So `node` needs to be a valid child of `parent`

Reply:
Ahh right, that damn API always confuses me 😔